src/k8s-extension/azext_k8s_extension/partner_extensions/OpenServiceMesh.py

```diff
@@ -93,3 +105,58 @@ def Update(self, extension, auto_upgrade_minor_version, release_train, version):
             release_train=release_train,
             version=version
         )
+
+
+def _validate_tested_distro(cmd, cluster_resource_group_name, cluster_name, extension_version):
+
+    if version.parse(str(extension_version)) <= version.parse("0.8.3"):
+        logger.warning('\"testedDistros\" field unavailable for version %s of osm-arc, '
+                       'cannot determine if this kubernetes distribution has been tested '
+                       'for osm-arc', extension_version)
+        return
+
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+    resources = cf_resources(cmd.cli_ctx, subscription_id)
+
+    cluster_resource_id = '/subscriptions/{0}/resourceGroups/{1}/providers/Microsoft.Kubernetes' \
+        '/connectedClusters/{2}'.format(subscription_id, cluster_resource_group_name, cluster_name)
+    try:
+        resource = resources.get_by_id(cluster_resource_id, '2020-01-01-preview')
+        cluster_distro = resource.properties['distribution'].lower()
+    except CloudError as ex:
+        raise ex
+
+    if cluster_distro == "general":
+        logger.warning('kubernetes distribution is \"general\", cannot determine if this kubernetes '
+                       'distribution has been tested for osm-arc')
+        return
+
+    tested_distros = _get_tested_distros(extension_version)
+
+    if tested_distros is None:
+        logger.warning('\"testedDistros\" field unavailable for version %s of osm-arc, '
+                       'cannot determine if this kubernetes distribution has been tested '
+                       'for osm-arc', extension_version)
+    elif cluster_distro in tested_distros.split():
+        logger.warning('%s is a tested kubernetes distribution for osm-arc', cluster_distro)
+    else:
+        logger.warning('%s is not a tested kubernetes distribution for osm-arc', cluster_distro)
+
+
+def _get_tested_distros(chart_version):
+
+    chart_arc = ChartBuilder({
+        "name": "osm-arc",
+        "version": str(chart_version),
+        "source": {
+            "type": "repo",
+            "location": "https://azure.github.io/osm-azure"
+        }
+    })
+    values = chart_arc.get_values()

```

Comment:
What happens if the passed chart version doesn't exist? It seems this would die with a fatal error

Reply:
If the chart version doesn't exist (ex for 0.8.5), it would give the following error: 

`pyhelm.repo.VersionError: Chart version 0.8.5 not found`

Reply:
@nshankar13 Can you please add a test to OpenServiceMesh to validate this error? Thanks :)